### PR TITLE
Teacher page

### PR DESCRIPTION
**Pull request by WhyHelloPear**
Teacher profile page is fully functional. Correctly displays summary stats for any given professor in the database and gives an overall calculated score.

Page also provides a full course catalog of the courses taught by said professor. Along with this, a pie chart is displayed to show the ratio of departments the teacher has taught in. 

NOTE: Further stats and graphs will be implemented in the future; this is just an update.

### CODE REVIEW

File: classes/management/commands/updateclasses.py [hunk 1]
```diff
@@ -221,59 +223,89 @@
...
     "EDU": "EDUC",
     "LAW": "LAWS",
+    "WMST": "WGST",
+    "JOUR": "JRNL",
 }
 
 
 def PopulateClasses():
-    Class.objects.all().delete()
+    # Class.objects.all().delete() # we won't need this after changes
```
[the remark this answers is not in the document]
Comment (WhyHelloPear) [reply classes/management/commands/updateclasses.py]: We don't need to delete the objects anymore. From now on, we check to see if the object already exists in the database and add it if it does not (if it does exist, we simply don't add it).

As for the bugs, I'll go through and double check. 

We can definitely reformat the page for looks.

File: classes/models.py [hunk 2]
```diff
@@ -61,15 +83,13 @@ class Class(models.Model):
...
+    schedule = models.ManyToManyField(Schedule, related_name="classes")
 
-    def empty_fields(self):
-        empty = []
-        for field in self.__dict__:
-            if self.__dict__[field] == None or self.__dict__[field] == "":
-                empty.append(field)
-        return empty
-
+    def in_schedule(self, user):  # broken for now
```
[the remark this answers is not in the document]
Comment (WhyHelloPear) [reply classes/models.py]: I didn't write that. That came from Jacob. My branch does not have a functioning class search/dsiplay.

File: classes/management/commands/updateclasses.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (WhyHelloPear) [reply classes/management/commands/updateclasses.py]: The chart definitely needs to have its own space. They overlap when the screen size is adjusted to anything other than fullscreen.

**Comment by jacobfelknor** [comment]
this request has been absorbed into others

**Pull request closed without merging**